library/src/main/java/com/android/mms/transaction/DownloadManager.java

```diff
@@ -68,13 +67,22 @@ public void downloadMultimediaMessage(final Context context, final String locati
                 .path(fileName)
                 .scheme(ContentResolver.SCHEME_CONTENT)
                 .build();
-        Intent download = new Intent(receiver.mAction);
+        Intent download = new Intent(receiver.ACTION);
+
         download.putExtra(MmsReceivedReceiver.EXTRA_FILE_PATH, mDownloadFile.getPath());
         download.putExtra(MmsReceivedReceiver.EXTRA_LOCATION_URL, location);
         download.putExtra(MmsReceivedReceiver.EXTRA_TRIGGER_PUSH, byPush);
         download.putExtra(MmsReceivedReceiver.EXTRA_URI, uri);
+
+        // originally this library added the uuid to the action each time it registered a receiver and broadcasted the download
+        // this would throw a 'sending non-protected broadcast' error that couldn't be resolved since we couldn't
+        // add the dynamic action name to the manifest
+        // setting a unique request code should be enough to uniquely identify the pending intent
+
+        int requestCode = UUID.randomUUID().hashCode();
```

Comment:
I don't quite understand what the unique requestCode does. Does it make sure that we keep broadcasting this instead of Android thinking the result has already been broadcasted?

Reply:
the original code added a uuid to the ACTION name each time it registered the receiver and broadcast the intent - we can assume they did this to make the intents unique

however, this would cause a 'sending non-protected broadcast' error, which we can't protect, since we have to explicitly define the action in the manifest

using a unique request code here keeps the same behaviour as originally intended, of making the intent unique. alternatively, we could've used intent.setData, but i found that doing this would cause the receiver `onReceive` to not get called

for additional information:
https://developer.android.com/reference/android/content/Intent#filterEquals(android.content.Intent)
https://stackoverflow.com/questions/21526319/whats-requestcode-used-for-on-pendingintent

Reply:
@aconanlai this comment was very enlightening. Can you include it in the comment?